router/internal/persistedoperation/operationstorage/cache.go

```diff
@@ -14,6 +16,25 @@ type OperationsCache struct {
 	Cache *ristretto.Cache[string, []byte]
 }
 
+func NewOperationsCache(cacheSize int64) (*OperationsCache, error) {
+	if cacheSize <= 0 {
+		return &OperationsCache{}, nil
+	}
+
+	cache, err := ristretto.NewCache(&ristretto.Config[string, []byte]{
```

Comment:
Ristretto needs to be closed otherwise it leaks goroutines. The client needs to be shutdown on server shutdown.

Reply:
Added to the close function